src/main/java/seedu/address/ui/MonthlyFeeListPanel.java

```diff
@@ -0,0 +1,119 @@
+package seedu.address.ui;
+
+import java.time.LocalDateTime;
+import java.util.logging.Logger;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ListChangeListener;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.scene.control.ListCell;
+import javafx.scene.control.ListView;
+import javafx.scene.layout.Region;
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.commons.util.DateUtil;
+import seedu.address.commons.util.FeeUtil;
+import seedu.address.model.fee.Month;
+import seedu.address.model.fee.MonthlyFee;
+import seedu.address.model.fee.Year;
+import seedu.address.model.student.Student;
+
+/**
+ * Panel containing the list of monthly fees for this month and the previous 2 month.
+ */
+public class MonthlyFeeListPanel extends UiPart<Region> {
+    private static final String FXML = "MonthlyFeeListPanel.fxml";
+    private final Logger logger = LogsCenter.getLogger(MonthlyFeeListPanel.class);
+
+    @FXML
+    private ListView<MonthlyFee> monthlyFeeListView;
+
+    /**
+     * Creates a {@code MonthlyFeeListPanel} with the given {@code ObservableList}.
+     */
+    public MonthlyFeeListPanel(ObservableList<Student> studentList) {
+        super(FXML);
+        ObservableList<MonthlyFee> monthlyFeeList = FXCollections.observableArrayList();
+
+        populateMonthlyFeeList(studentList, monthlyFeeList);
+        System.out.println("Size is: " + monthlyFeeList.size());
+
+        studentList.addListener((ListChangeListener<Student>) change -> {
+            while (change.next()) {
+                monthlyFeeList.clear();
+                populateMonthlyFeeList(studentList, monthlyFeeList);
+                populateMonthlyFeeListView(monthlyFeeList);
+            }
+        });
```

Comment:
I think can do what Samuel did and abstract this out to an addListener() method.
I think it makes it look neater. I will probably do this for my components as well.

Reply:
Ah right! It will make the code looks nicer ah. Have done so in the latest commit! Thanks!